AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/IncomingAsyncSubscriptionEventPublisher.swift

```diff
@@ -42,7 +42,10 @@ final class IncomingAsyncSubscriptionEventPublisher: Cancellable {
     private let incomingSubscriptionEvents: PassthroughSubject<Event, DataStoreError>
     private let awsAuthService: AWSAuthServiceBehavior
 
-    init(modelSchema: ModelSchema, api: APICategoryGraphQLBehavior, auth: AuthCategoryBehavior?,
+    init(modelSchema: ModelSchema,
+         api: APICategoryGraphQLBehavior,
+         configuration: DataStoreConfiguration,
```

Comment:
as above, we shouldn't be passing the entire configuration object

Reply:
> The overall design makes sense, but I'm not joyful about the growth of the DataStoreConfiguration interface, and I don't like passing that whole configuration object around to the underlying components. Comments inline

Regrading the configuraiton object being unwieldy, that is the worst case scenario.  For customers only wanting to specify a selective sync, it would look something like this:

```
            let syncExpression1 = DataStoreSyncExpression.syncExpression(Todo.schema, where: {
                Todo.keys.tag == AppDelegate.tagToSearch
            })
            let dataStorePlugin = AWSDataStorePlugin(modelRegistration: AmplifyModels(),
                                                     configuration: .custom(syncExpressions: [syncExpression1]))
```

Are you still concerned about bloating this interface?